File: Scadex.Model/Dtos/Camera/Queries/CameraDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.Camera.Queries;

public class CameraDto : IDto
{
    public Guid Id { get; set; }
    public Guid CabinetId { get; set; }
    public string? CabinetName { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public string? Manufacturer { get; set; }
    public string? Model { get; set; }

    public string IpAddress { get; set; } = null!;
    public int RtspPort { get; set; }
    public int HttpPort { get; set; }
    public int? HttpsPort { get; set; }

    public string? Username { get; set; }

    public string? Password { get; set; }

    public int MainStreamChannel { get; set; }
    public int SubStreamChannel { get; set; }
    public bool MainStreamEnabled { get; set; }
    public bool SubStreamEnabled { get; set; }
    public int SnapshotChannel { get; set; }

    public int? MonitoringPort { get; set; }
    public int? DeviceStatusId { get; set; }

    public string? DeviceStatusName { get; set; }
    public DateTime? LastSeen { get; set; }
    public int PingIntervalSec { get; set; }
    public bool IsMonitoringEnabled { get; set; }
    public string? LastConnectionError { get; set; }

    #region --- IAuditableEntity ---
    public DateTime? CreateDateUtc { get; set; }
    public DateTime? UpdateDateUtc { get; set; } 
    #endregion

    #region --- IActivatableEntity ---
    public bool IsActive { get; set; } 
    #endregion
}

File: Scadex.WebUI/src/models/camera/queries/cameraDto.ts
/**
 * Ayna: Scadex.Model/Dtos/Camera/Queries/CameraDto.cs
 * Sözleşme: docs/api-contract/11-camera.md
 */
export interface CameraDto {
  id: string;
  cabinetId: string;
  cabinetName: string | null;
  name: string;
  description: string | null;
  manufacturer: string | null;
  model: string | null;

  ipAddress: string;
  rtspPort: number;
  httpPort: number;
  httpsPort: number | null;

  username: string | null;
  /**
   * Kamera parolası — DÜZ METİN olarak döner (kullanıcı kararı).
   *
   * Önceki sözleşmede yerine `hasPassword: boolean` gidiyordu; sistem kapalı
   * ağda çalıştığı ve bu aşamada kamera kimlik bilgilerinin gizlenmesi
   * istenmediği için doğrudan döndürülüyor.
   *
   * Yazma yolu YİNE ÜÇ DURUMLU: `undefined` = dokunma, `''` = sil, dolu = değiştir.
   */
  password: string | null;

  mainStreamChannel: number;
  subStreamChannel: number;
  mainStreamEnabled: boolean;
  subStreamEnabled: boolean;
  snapshotChannel: number;

  monitoringPort: number | null;
  /**
   * `DeviceStatus` lookup FK — Device/Cabinet ile AYNI sözlük.
   * `null` = **hiç yoklanmadı**; `Offline` (0) ile aynı şey DEĞİL.
   */
  deviceStatusId: number | null;
  deviceStatusName: string | null;
  lastSeen: string | null;
  pingIntervalSec: number;
  isMonitoringEnabled: boolean;
  lastConnectionError: string | null;

  isActive: boolean;
  createDateUtc: string | null;
  updateDateUtc: string | null;
}
